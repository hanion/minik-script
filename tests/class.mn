// class

Animal :: class {
	name: string;

	Animal :: (name) {
		this.name = name;
	}

	print_name :: () {
		print this.name;
	}
}

print Animal;
dog := Animal("doug");
print dog;
print dog.name;
print dog.print_name;
dog.print_name();
dog.name = "dawg";
dog.print_name();


cat := Animal("kat");
cat.print_name();
cat.name = "kath";
cat.print_name();


Animal("fish").print_name();

