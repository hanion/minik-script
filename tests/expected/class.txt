<class Animal>
<instance of Animal>
doug
<fn print_name>
doug
dawg
kat
kath
fish
